ui/components/app/advanced-gas-popover/advanced-gas-popover.js

```diff
@@ -0,0 +1,319 @@
+          toDenomination: 'GWEI',
+        });
+        setMaxBaseFeeMultiplier(value);
+        setMaxBaseFeeGWEI(baseFeeValue);
+      }
+    },
+    [
+      estimatedBaseFee,
+      editingInGwei,
+      setMaxBaseFeeGWEI,
+      setMaxBaseFeeMultiplier,
+    ],
+  );
+
+  const updateBaseFeeMode = useCallback(
+    (value) => {
+      if (value) {
+        const baseFeeGwei = multiplyCurrencies(
+          estimatedBaseFee,
+          maxBaseFeeMultiplier,
+          {
+            numberOfDecimals: 6,
+            multiplicandBase: 16,
+            multiplierBase: 10,
+            toNumericBase: 'dec',
+            fromDenomination: 'WEI',
+            toDenomination: 'GWEI',
+          },
+        );
+        setMaxBaseFeeGWEI(baseFeeGwei);
+      } else {
+        const baseFeeMul = divideCurrencies(
+          decGWEIToHexWEI(maxBaseFeeGWEI),
+          estimatedBaseFee,
+          {
+            numberOfDecimals: 6,
+            dividendBase: 16,
+            divisorBase: 16,
+            toNumericBase: 'dec',
+          },
+        );
+        setMaxBaseFeeMultiplier(baseFeeMul);
+      }
+      setEditingInGwei(value);
+    },
+    [
+      estimatedBaseFee,
+      maxBaseFeeMultiplier,
+      maxBaseFeeGWEI,
+      setMaxBaseFeeGWEI,
+      setMaxBaseFeeMultiplier,
+      setEditingInGwei,
+    ],
+  );
+
+  return (
+    <Popover
+      className="advanced-gas-popover"
+      title={t('advancedGasFeeModalTitle')}
+      onBack={() => onClose()}
+      onClose={() => onClose()}
+    >
+      <Box className="advanced-gas-popover" margin={4}>
+        <FormField
+          onChange={(value) => {
+            onManualChange?.();
+            setBaseFee(value);
+          }}
+          titleFontSize={TYPOGRAPHY.H7}
+          titleText={t('maxBaseFee')}
+          titleUnit={
+            editingInGwei ? t('gweiInParanthesis') : t('multiplierInParathesis')
+          }
+          tooltipText={t('advancedBaseGasFeeToolTip')}
+          titleDetail={
+            editingInGwei ? (
+              <Button
+                className="advanced-gas-popover__edit-link"
+                type="link"
+                onClick={() => updateBaseFeeMode(false)}
+              >
+                <I18nValue messageKey="editInMultiplier" />
+              </Button>
+            ) : (
+              <Button
+                className="advanced-gas-popover__edit-link"
+                type="link"
+                onClick={() => updateBaseFeeMode(true)}
+              >
+                <I18nValue messageKey="editInGwei" />
+              </Button>
+            )
+          }
+          value={editingInGwei ? maxBaseFeeGWEI : maxBaseFeeMultiplier}
+          detailText={
+            editingInGwei
+              ? t('baseFeeFiatFromMultiplier', [
+                  maxBaseFeeMultiplier,
+                  baseFeeFiatMessage,
+                ])
+              : t('baseFeeFiatFromGwei', [baseFee, baseFeeFiatMessage])
+          }
+          numeric
```

Comment:
I notice that these are numeric fields. I know we have been using numeric fields for gas fees so far, but I feel like the way we use them right now is not ideal and has lead to a really quirky and sometimes frustrating experience. See: https://github.com/MetaMask/metamask-extension/issues/11876 and https://consensys.slack.com/archives/GTQAGKY5V/p1631118586253200. I know that we are not unified on this opinion, so maybe we would need to have further discussion on this, but I wonder if it's worth it to take the time this time around to explore a different solution for these kinds of fields? I can post more of my thoughts on this if need be.

Reply:
I have experienced this issue as well, but at the moment I am using the FormField for the input and label, changing it there would be a better solution I think (maybe as a different PR). It would be better to get input from the rest of the team as well.

Reply:
Yeah, I think we'll eventually want to fix this in order to support the full 18 decimal places for wei, but I agree that this can be done in a separate PR.